vscode-dotnet-runtime-library/src/Utils/WebRequestWorker.ts

```diff
@@ -73,12 +77,50 @@ export class WebRequestWorker
             }
         });
 
-            this.client = setupCache(uncachedAxiosClient,
-                {
-                    storage: buildMemoryStorage(),
-                    ttl: this.cacheTimeToLive
-                }
-            );
+        // Record when the web requests begin:
+        // Register this so it happens before the cache https://axios-cache-interceptor.js.org/guide/interceptors#explanation
+        uncachedAxiosClient.interceptors.request.use( req => {
+            req.headers.startTime = process.hrtime.bigint();
```

Comment:
I'm skittish about setting a header - this will be sent to a server. You should be able to modify the request object directly and set a field on it to retrieve later. e.g.:

```
// in the request interceptor
req.timings.start = process.hrtime.bigint();
....
// in the response interceptor
let end = process.hrtime.bigint();
let delta = end - res.config.timings.start;
```

If this works we don't leak any other data to any server.

Reply:
Fantastic observation, thank you for catching that.

I tried to do this the traditional way before headers, but Axios seems to have broken the ability to modify the object directly https://github.com/axios/axios/issues/5573. And their documentation kinda sucks, or maybe I didnt search it well enough. Now it looks like you can have a lot of wrapper methods and interfaces or you can change the data or config and store it there. So ill try that.

Reply:
Ok maybe Im silly. Idk. Maybe this was all very JS-centric and its just always been a typescript thing where you have to make a bunch of wrapper types and functions to make the compiler not mad. I'm just going to make it a bunch of any's.